Problem: cargo test fails to compile with `grpc` feature

`cargo test --features grpc`

```
error[E0432]: unresolved import `chainlib::message::Transfer`
 --> examples/example_amino.rs:7:5
  |
7 | use chainlib::message::Transfer;
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^ no `Transfer` in `message`

error[E0061]: this function takes 2 arguments but 1 argument was supplied
  --> examples/example_amino.rs:15:18
   |
15 |     let client = Client::new(base_api_url);
   |                  ^^^^^^^^^^^ ------------ supplied 1 argument
   |                  |
   |                  expected 2 arguments

error[E0308]: mismatched types
  --> examples/example_amino.rs:28:67
   |
28 |     let mut builder = TxBuilder::new(key_service, chain_id, memo, Some(fee.clone()), gas);
   |                                                                   ^^^^^^^^^^^^^^^^^ expected `u64`, found enum `Option`
   |
   = note: expected type `u64`
              found enum `Option<Amount>`

error[E0308]: mismatched types
  --> examples/example_amino.rs:28:86
   |
28 |     let mut builder = TxBuilder::new(key_service, chain_id, memo, Some(fee.clone()), gas);
   |                                                                                      ^^^ expected struct `chainlib::proto::cosmos::tx::v1beta1::Fee`, found integer
   |
   = note: expected enum `Option<chainlib::proto::cosmos::tx::v1beta1::Fee>`
              found enum `Option<{integer}>`

error[E0061]: this function takes 0 arguments but 1 argument was supplied
  --> examples/example_amino.rs:40:22
   |
40 |     let tx = builder.build(SyncMode::Sync).await.unwrap();
   |                      ^^^^^ -------------- supplied 1 argument
   |                      |
   |                      expected 0 arguments

error[E0308]: mismatched types
  --> examples/example_amino.rs:41:40
   |
41 |     let response = client.broadcast_tx(tx).await?;
   |                                        ^^ expected struct `Vec`, found struct `std::string::String`
   |
   = note: expected struct `Vec<u8>`
              found struct `std::string::String`

error[E0277]: the `?` operator can only be applied to values that implement `Try`
  --> examples/example_amino.rs:41:20
   |
41 |     let response = client.broadcast_tx(tx).await?;
   |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the `?` operator cannot be applied to type `ResponseBroadcastTx`
   |
   = help: the trait `Try` is not implemented for `ResponseBroadcastTx`
   = note: required by `into_result`

error[E0698]: type inside `async` block must be known in this context
  --> examples/example_amino.rs:18:9
   |
18 |     let gas = Some(300000);
   |         ^^^ cannot infer type for type `{integer}`
   |
note: the type is part of the `async` block because of this `await`
  --> examples/example_amino.rs:41:20
   |
41 |     let response = client.broadcast_tx(tx).await?;
   |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 8 previous errors

Some errors have detailed explanations: E0061, E0277, E0308, E0432, E0698.
For more information about an error, try `rustc --explain E0061`.
error: could not compile `chainlib`
```

--- a/.github/workflows/chainlib.yml
+++ b/.github/workflows/chainlib.yml
@@ -63,6 +63,8 @@ jobs:
           target: ${{ matrix.target }}
           override: true
       - run: cargo build --target ${{ matrix.target }} --release --all-features
+      - run: cargo build --target ${{ matrix.target }} --release --example amino
+      - run: cargo build --target ${{ matrix.target }} --release --example protobuf --features=grpc
 
   test:
     runs-on: ubuntu-latest
@@ -79,4 +81,4 @@ jobs:
           profile: minimal
           toolchain: ${{ matrix.rust }}
           override: true
-      - run: cargo test --release
+      - run: cargo test --release --lib --all-features

--- a/README.md
+++ b/README.md
@@ -35,3 +35,6 @@ chain-maind tx bank send \
 or 
 
 `cargo build --example protobuf --features=grpc`
+
+# unit test
+`cargo test --lib --all-features`

